packages/core/repl/repl-context.ts

```diff
@@ -1,87 +1,38 @@
-import {
-  DynamicModule,
-  INestApplication,
-  InjectionToken,
-  Logger,
-  Type,
-} from '@nestjs/common';
-import { clc } from '@nestjs/common/utils/cli-colors.util';
+import { INestApplication, InjectionToken, Logger } from '@nestjs/common';
 import { ApplicationConfig } from '../application-config';
 import { ModuleRef, NestContainer } from '../injector';
 import { InternalCoreModule } from '../injector/internal-core-module';
 import { Module } from '../injector/module';
-import { MetadataScanner } from '../metadata-scanner';
+import {
+  DebugReplFn,
+  GetReplFn,
+  HelpReplFn,
+  MethodsReplFn,
+  ResolveReplFn,
+  SelectReplFn,
+} from './native-functions';
+import type { ReplFunctionClass } from './repl.interfaces';
 
 type ModuleKey = string;
-type ModuleDebugEntry = {
+export type ModuleDebugEntry = {
   controllers: Record<string, InjectionToken>;
   providers: Record<string, InjectionToken>;
 };
 
 export class ReplContext {
-  private debugRegistry: Record<ModuleKey, ModuleDebugEntry> = {};
-  private readonly container: NestContainer;
-  private readonly logger = new Logger(ReplContext.name);
-  private readonly metadataScanner = new MetadataScanner();
+  public nativeFunctions: InstanceType<ReplFunctionClass>[];
 
-  constructor(private readonly app: INestApplication) {
+  public debugRegistry: Record<ModuleKey, ModuleDebugEntry> = {};
+  public readonly container: NestContainer;
+  public readonly logger = new Logger(ReplContext.name);
```

Comment:
```suggestion
  public debugRegistry: Record<ModuleKey, ModuleDebugEntry> = {};
  public readonly container: NestContainer;
  public readonly logger = new Logger(ReplContext.name);
```
why all these props are public now?

Reply:
I didn't like that but:

- `debugRegistry` is public because `DebugReplFn#action` depends on it
- `logger` is public because `ReplFunction` uses it as a value for their `logger` field (but allows childs of that class to have their own logger)
- `container` is private now